packages/analytics/get-product-intent-from-url/index.ts

```diff
@@ -0,0 +1,29 @@
+const products = [
+  'packer',
+  'terraform',
+  'vault',
+  'boundary',
+  'consul',
+  'nomad',
+  'waypoint',
+  'vagrant',
+] as const
+
+type Product = typeof products[number]
+
+export const getProductIntentFromURL = (url?: string): Product | null => {
+  if (!url && typeof window === 'undefined') {
+    return null
+  }
+  const fromUrl = url || window.location.href.toString()
+  const result: Partial<Record<Product, number>> = {}
+  products.forEach((product) => {
+    let index = fromUrl.indexOf(product)
+    result[product] = index
+  })
+  const productIntent = (Object.entries(result) as [Product, number][])
+    .filter(([_, index]) => index > -1)
+    .sort((a, b) => a[1] - b[1])
+
+  return productIntent.length ? productIntent[0][0] : null
+}
```

Comment:
A quick comment above this block would help to understand the logic, which I believe to be:
* Build a map of product -> index it first appears in the URL
* Take the product which appears first in the URL

Reply:
Added https://github.com/hashicorp/web-platform-packages/pull/63/commits/cd3fece10c0a714c665b3df339c1f171bea4aef4